Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamerBase.cpp

```diff
@@ -484,6 +470,14 @@ bool MediaPlayerPrivateGStreamerBase::handleSyncMessage(GstMessage* message)
             GstStructure* contextStructure = gst_context_writable_structure(context.get());
             gst_structure_set(contextStructure, "decryption-system-id", G_TYPE_STRING, preferredKeySystemUuid, nullptr);
             gst_element_set_context(GST_ELEMENT(GST_MESSAGE_SRC(message)), context.get());
+#if USE(OPENCDM)
+            if (!findAndSetPendingProtectionEventByInitData(initDataChunks, keySystemProtectionEventMap.get(preferredKeySystemUuid))) {
```

Comment:
You can do all this code inside `findAndSetPendingProtectionEventByInitData`. Actually, you can rename it to `addPendingProtectionEventToInitDataMapping`.

Reply:
Sure we will update it, thank you